test/e2e/common/configmap.go

```diff
@@ -134,6 +134,27 @@ var _ = ginkgo.Describe("[sig-node] ConfigMap", func() {
 		configMap, err := newConfigMapWithEmptyKey(f)
 		framework.ExpectError(err, "created configMap %q with empty key in namespace %q", configMap.Name, f.Namespace.Name)
 	})
+
+	ginkgo.It("should patch ConfigMap successfully", func() {
+		name := "configmap-test-" + string(uuid.NewUUID())
+		configMap := newConfigMap(f, name)
+		configMapOriginalState := *configMap
+		ginkgo.By(fmt.Sprintf("Creating configMap %v/%v", f.Namespace.Name, configMap.Name))
+		_, err := f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Create(configMap)
+		framework.ExpectNoError(err)
+
+		configMap.Data = map[string]string{
+			"data": "value",
+		}
+		ginkgo.By(fmt.Sprintf("Updating configMap %v/%v", f.Namespace.Name, configMap.Name))
+		_, err = f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Update(configMap)
+		framework.ExpectNoError(err)
+
+		configMapFromUpdate, err := f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Get(name, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		ginkgo.By(fmt.Sprintf("Verifying update of configMap %v/%v", f.Namespace.Name, configMap.Name))
+		framework.ExpectNotEqual(configMapFromUpdate.Data, configMapOriginalState.Data)
```

Comment:
Why don't we check the `Get()` data is the same as `Update()`d data(`"data": "value"`) here?
I mean it is nice to add `framework.ExpectEqual(configMapFromUpdate.Data, configMap.Data)`

Reply:
configMap.Data's value is changed, so it doesn't make sense to compare against it.

[test/e2e/common/configmap.go: L146-148](https://github.com/ii/kubernetes/blob/b10743236ef08ab8cf4b3cf5ac4ea9911db453fa/test/e2e/common/configmap.go#L146)
```golang
 		configMap.Data = map[string]string{
			"data": "value",
		}
```

Reply:
I am not sure I can follow what you are saying here.
In this test, the config map is updated with
```
 		configMap.Data = map[string]string{
			"data": "value",
		}
```
So the config map date which is gotten at line 153 should be the same as the above value, isn't it?

Reply:
Perhaps the purpose of the variables is being confused.
`configMap` is the original configMap value.   
`configMapOriginalState` has the value of `configMap` before it's changed.  
`configMapFromUpdate` has the value of `configMap` after it's updated, pulling it from it's state in the cluster -- must be different from `configMapOriginalState`.

Reply:
> configMapFromUpdate has the value of configMap after it's updated, pulling it from it's state in the cluster -- must be different from configMapOriginalState.

Yes right. Then it is better to check not only difference from the original but also that the updated data should be the same as the data which is used for updating (`"data": "value"`)

Reply:
Updating the Expect statement in 7218182f119.